src/rust/engine/process_execution/docker/src/docker.rs

```diff
@@ -218,9 +218,10 @@ async fn credentials_for_image(
                 // TODO: https://github.com/keirlawson/docker_credential/issues/7 means that this will only
                 // work for credential helpers and credentials encoded directly in the docker config,
                 // rather than for general credStore implementations.
-                let credential = docker_credential::get_credential(&server).map_err(|e| {
-                    format!("Failed to retrieve credentials for server `{server}`: {e}")
-                })?;
+                let Ok(credential) = docker_credential::get_credential(&server) else {
+                    log::warn!("Failed to retrieve credentials for server `{}`", server);
```

Comment:
Maybe:
- s/credentials/Docker credentials/ ?
- Output the error in the log message?